### ISSUE TITLE
Device context not being properly returned for certain android models

### ISSUE THREAD

**Issue by Fwang36** [getsentry/sentry#50137]
### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

Several customers have reported certain Android models are not properly returning device context.  Here is the portion of the event JSON from a Samsung Galaxy S22 on Android 13 and Pixel 4 on Android 13

<img width="568" alt="Screenshot 2023-05-31 at 1 18 31 PM" src="https://github.com/getsentry/sentry/assets/96670625/1387f4ad-3b3c-4a2e-b0bf-bf6908b60459">

[Example Event From My Sentry Org](https://francisorg.sentry.io/issues/4219135781/events/b88cdb889f544286a78ca329ccd7295e/?query=is%3Aunresolved&referrer=next-event&stream_index=0). (updated event with data scrubbing turned off)

### Expected Result

Returns Proper Device Context

### Actual Result

Returns generic device of "K"

### Product Area

Other

**Comment by malwilley** [comment]
It looks like this is a result of Chrome's recent changes to user-agent on Android devices: https://developer.chrome.com/blog/user-agent-reduction-android-model-and-version/

This will not be a simple fix since we will need to collect device information through means other than the user-agent. Will bring this up internally to see if there is a plan for this.

**Comment by mydea** [comment]
Possibly related: https://github.com/getsentry/relay/issues/1183

**Comment by mustafa0x** [comment]
I'm unable to debug this issue since a device name isn't set. How can I retrieve it?

<img width="743" alt="Image" src="https://github.com/user-attachments/assets/e47309bd-5fee-43d1-a74d-942d3b1e9efd" />

<hr>

<img width="341" alt="Image" src="https://github.com/user-attachments/assets/1da90203-ff24-4144-b0bf-99b634fe04f5" />

**Comment by mrduncan** [comment]
@mustafa0x the JSON for that event (in either the "{}" or "JSON" buttons, depending on which version of the UI you are using) will have all of the available information for the event.

If you can share a link to the event itself here we can see if there are other properties we might be able to extract by default.

**Comment by lobsterkatie** [comment]
Yeah, it's a reasonable thing to want to know. I'm going to defer to the folks on @getsentry/team-mobile on this one, because they will know best if there's any other place from which to draw the relevant data.

**Comment by kahest** [comment]
@lobsterkatie afaict this event (same as described in this issue) is sent from a JS SDK running in a website on an Android browser. Looking at the JSON, I don't see more useful device info anywhere else in the event. Maybe @mydea might be able to help out here, but it seems like it's the same issue described by @malwilley [above](https://github.com/getsentry/sentry/issues/50137#issuecomment-1572606019).
